discovery-provider/src/api_helpers.py

```diff
@@ -1,17 +1,54 @@
-from flask import jsonify
+import logging
+import json
+import datetime
 import redis
+from web3 import Web3
+from web3.auto import w3
+from eth_account.messages import encode_defunct
+from flask import jsonify
+
+from src.utils import helpers
 from src.utils.config import shared_config
 from src.utils.redis_constants import latest_block_redis_key, most_recent_indexed_block_redis_key
 
 redis_url = shared_config["redis"]["url"]
 redis = redis.Redis.from_url(url=redis_url)
+logger = logging.getLogger(__name__)
+disc_prov_version = helpers.get_discovery_provider_version()
 
+METADATA_FIELDS = ['success', 'latest_indexed_block', 'latest_chain_block']
+API_SIGNING_FIELDS = ['timestamp', 'signature']
+
+# Subclass JSONEncoder
+class DateTimeEncoder(json.JSONEncoder):
+    # Override the default method
+    def default(self, o): # pylint: disable=E0202
+        if isinstance(o, (datetime.date, datetime.datetime)):
+            # TODO - the Z is required in JS date format
+            return o.isoformat() + " Z"
+        return json.JSONEncoder.default(self, o)
 
 def error_response(error, error_code=500):
     return jsonify({'success': False, 'error': error}), error_code
 
+# Create a response dict with just data, signature, and timestamp
+# This response will contain a duplicate of response_entity
+def success_response_backwards_compat(response_entity=None, status=200):
+    data = {'data': response_entity}
```

Comment:
in sprit of the pure function argument, you could simplify this function to be. i personally find this much clearer to read

```python3
def success_response_backwards_compat(response_entity=None, status=200):
    response_dictionary = {**response_entity, 'data': response_entity}
    signature, timestamp = generate_signature_and_timestamp(response_dictionary['data'])
    response_dictionary['signature'] = signature
    response_dictionary['timestamp'] = timestamp
    return jsonify(response_dictionary), status
```